sdk/test-utils/recorder/src/utils.node.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import fs from "fs-extra";
+import nock from "nock";
+
+/**
+ * Returns an array of `nock.Interceptor`s capable of matching the HTTP methods with the regex path provided
+ *
+ * @param {nock.Scope} nockScope
+ * @param {RegExp} regex
+ * @returns {Array<nock.Interceptor>}
+ */
+function getNockInterceptors(nockScope: nock.Scope, regex: RegExp): Array<nock.Interceptor> {
+  return [
+    nockScope.get(regex),
+    nockScope.post(regex),
+    nockScope.put(regex),
+    nockScope.head(regex),
+    nockScope.patch(regex),
+    nockScope.merge(regex),
+    nockScope.delete(regex),
+    nockScope.options(regex)
+  ];
+}
+
+/**
+ * Meant only for node tests during playback mode
+ * Delay and reply with errors for the new requests with the base urls from the existing current recording
+ *
+ * @export
+ * @param {*} nock
+ * @param {string} recordingPath
+ */
+export function fakeOutDelaysForNewRequests(nock: any, recordingPath: string) {
+  const matchAll = require("match-all");
+
+  // Read the content from the recording
+  const content = fs.readFileSync(recordingPath, "utf-8").toString();
+
+  // Find all the mocked urls in the recording
+  const setOfUrlsInRecording: Set<string[]> = new Set();
+  Array.from(
```

Comment:
This confuses me a little, since it seems like we're using Array.from to call map() via its second arg, but the first arg is presumably already an Array (due to the toArray call).

It also feels weird to use a map function but not care about the result. Like Array.from returns a new array that is discarded because we used side effects in the map function.

Since the Set() constructor can take an iterable, why not do something like this?

```ts
conts matchArray = matchAll(content, /nock\('(.*?)', {"encodedQueryParams":true}\)/g).toArray();
const setOfUrlsInRecording: Set<string[]> = new Set(matchArray);
```

Reply:
Addressed!